pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -6035,3 +6035,308 @@ func createKafkaJbpmObject(dateFormat string, tasksTopics string, casesTopics st
+	assert.True(t, corsEnabled)
+	assert.NotNil(t, cr.Spec.Objects.Dashbuilder)
+	//cors := cr.Status.Applied.Objects.Dashbuilder.Cors
+	//assert.NotNil(t, cors)
+	//checkCors(t, cors)
+}
+
+func isCORSEnabled(envs []corev1.EnvVar) bool {
+	corsEnabled := false
+	for _, env := range envs {
+		if strings.HasPrefix(env.Name, "AC_ALLOW") || strings.HasPrefix(env.Name, "FILTERS") {
+			corsEnabled, _ = strconv.ParseBool(env.Value)
+		}
+	}
+	return corsEnabled
+}
+
+func TestEnvironmentWithoutCORS(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-no-cors",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhdmTrial,
+			Objects: api.KieAppObjects{
+				Console: &api.ConsoleObject{},
+				Servers: []api.KieServerSet{},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting test-no-cors Test environment")
+	assert.NotNil(t, env)
+	corsEnabledKieServer := isCORSEnabled(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env)
+	assert.False(t, corsEnabledKieServer)
+	assert.Empty(t, cr.Spec.Objects.Servers)
+	assert.Nil(t, cr.Status.Applied.Objects.Servers[0].Cors)
+	corsEnabledKieConsole := isCORSEnabled(env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env)
+	assert.False(t, corsEnabledKieConsole)
+	assert.Empty(t, cr.Spec.Objects.Console)
+	assert.Nil(t, cr.Status.Applied.Objects.Console.Cors)
+}
+
+func TestEnvironmentWithCORS(t *testing.T) {
+	const name = "test-cors"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamTrial,
+			Objects: api.KieAppObjects{
+				Console: &api.ConsoleObject{
+					Cors: getDefaultCors(),
+				},
+				Servers: []api.KieServerSet{
+					{
+						Cors: getDefaultCors(),
+					},
+				},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting test-cors Test environment")
+	assert.NotNil(t, env)
+	checkKieServerCORSAssertions(t, cr, env)
+	cors := cr.Status.Applied.Objects.Servers[0].Cors
+	assert.NotNil(t, cors)
+	checkCors(t, cors)
+	checkConsoleCORSAssertions(t, cr, env)
+}
+
+func TestEnvironmentWithPartialCORS(t *testing.T) {
+	const name = "test-cors"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamTrial,
+			Objects: api.KieAppObjects{
+				Console: &api.ConsoleObject{
+					Cors: getPartialCors(),
+				},
+				Servers: []api.KieServerSet{
+					{
+						Cors: getPartialCors(),
+					},
+				},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting test-cors Test environment")
+	assert.NotNil(t, env)
+	checkKieServerCORSAssertions(t, cr, env)
+	cors := cr.Status.Applied.Objects.Servers[0].Cors
+	assert.NotNil(t, cors)
+	checkCustomCors(t, cors)
```

Comment:
When setting partial CORS, the parameters that are not being set should be empty, according to the tests, it is not happening,

Reply:
@spolti  We are setting the defaults if the params are empty https://github.com/kiegroup/kie-cloud-operator/pull/634/files#diff-29ad51b906bc49dd54db094a3b626e8e4a04628d03562e6c0d45db0e391ac2fbR1818-R1861 otherwise the CORS configuration is broken.